### ISSUE TITLE
Update guidance for use of the message-aria-describedby by prop

### ISSUE THREAD

**Issue by caw310** [department-of-veterans-affairs/vets-design-system-documentation#2012]
## Description
Jami added a new prop message-aria-describedby to several different form components. The guidance for these components should be updated with description of use for the message-aria-describedby prop.
va-checkbox
va-number-input
va-text-input
va-textarea

## Details
Tickets for reference
https://github.com/department-of-veterans-affairs/va.gov-team/issues/24502 
https://github.com/department-of-veterans-affairs/component-library/pull/788

## Tasks
- [ ] Update guidance for va-checkbox, va-number-input, va-text-input, va-textarea

## Acceptance Criteria
- [ ] Guidance on design.va.gov for va-checkbox, va-number-input, va-text-input, va-textarea has been updated

**Comment by humancompanion-usds** [comment]
If @rsmithadhoc can capture what this guidance should say here in this ticket then I can put it onto the site.

**Comment by rsmithadhoc** [comment]
@humancompanion-usds 

Here is the guidance for that:

> In HTML, the attribute `aria-describedby` accepts `id`s of the elements that describe an object. This is used to establish a relationship between an element and text elsewhere that describes it for screen readers.

> However, the VA.gov Design System uses Web Components and the Shadow DOM, which prevents HTML's `aria-describedby` from being able to establish the relationship between elements. Because of that,`message-aria-describedby` is used in our components instead. Instead of accepting `id`s, it accepts a string with message to read out. This message is placed inside the Shadow DOM, hidden visually, but made accessible to screen readers. This allows it to function similarly to `aria-describedby` and read out the descriptive text when the element is focused.
